lib/order.rb

```diff
@@ -0,0 +1,49 @@
+require_relative './takeaway'
+
+class Order 
+  attr_reader :order_list, :takeaway
+
+  def initialize(takeaway = Takeaway.new)
+    @order_list = []
+    @takeaway = takeaway
+  end
+
+  def add_by_name(name)
+    @takeaway.menu_hash.each do |item_name, price|
+      if name == item_name
+        add_item({ item_name => price })
+      end
+    end
+  end
+
+  def print_current_order
+    total = 0
+    @order_list.each do |item|
+      puts "#{item.keys.first} : £%.2f" % item.values.first
+      total += item.values.first
+    end
+    puts "Total : £%.2f" % total
```

Comment:
It could be worth moving this to another class to handle the money related behaviours of the order

Reply:
I agree with @GraemeHedges – I'd separate things that are involved in printing / displaying from the business logic.